src/state/predictions/hooks.ts

```diff
@@ -136,6 +138,31 @@ export const useGetOrFetchLeaderboardAddressResult = (account: string) => {
   return addressResult
 }
 
-export const useGetSelectedAddress = () => {
-  return useSelector((state: State) => state.predictions.leaderboard.selectedAddress)
+export const useGetSelectedAddress = (): string => {
+  return useSelector((state: PredictionsState) => state.leaderboard.selectedAddress)
+}
+
+// Because Modal Component is rendered outside the Prediction Page contexts
+// We have to pass local state as props instead of retrieving directly in component
+export const useStatModalProps = (account?: string) => {
```

Comment:
umm can we inject the provider in _app.tsx?

Reply:
we couldn't achieve various prediction config if Injecting Config Provider in `_app.tsx`. The flaw is in the Modal component design. It should use Portal to keep the Context.